lib/phoenix_swagger.ex

```diff
@@ -237,6 +237,10 @@ defmodule PhoenixSwagger do
     |> Map.from_struct()
     |> to_json()
   end
+  def to_json(%{type: typ, '$ref': ref}) when is_nil(typ) and is_nil(ref), do: :null
```

Comment:
This feels like it is fixing a symptom rather than the cause of the bug.

From the [jsonapi spec](http://jsonapi.org/format/#document-compound-documents)

> In a compound document, all included resources MUST be represented as an array of resource objects in a top-level included member.

Maybe try adding some properties to the schema at https://github.com/xerions/phoenix_swagger/blob/master/lib/phoenix_swagger/json_api.ex#L123

?